Add onboarding guide for SplashKit Expansion

matt439 opened the pull request:
# Description

Setting up the development environment for the splashkit-core repository can be confusing and daunting. This guide aims to help students start contributing to splashkit-core faster. I was inspired by my own personal difficulties as a junior, and added tips that would have helped me greatly.

Despite having been taught Git in SIT223, I found myself struggling to navigate the terminology when I joined SplashKit. Therefore, I have included many Git instructions in the guide. If students are confident with Git, they can simply skip over those parts.

Variations of this guide can be produced in the future detailing the specific instructions for MacOS, Linux, and MSYS2.

## Type of change

- [x] Documentation (update or new)

## How Has This Been Tested?

Ran Prettier and Vale linters.

## Testing Checklist

- [x] Tested in latest Chrome
- [x] Tested in latest Safari
- [x] Tested in latest Firefox

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

breezy-codes commented on src/content/docs/Products/SplashKit/Splashkit Expansion/Documentation/onboarding-guide.mdx:
```diff
@@ -0,0 +1,205 @@
+This guide will cover all the steps required to get contributing to the splashkit-core repository.
+Feel free to skip steps which you have already completed or are familiar with.
+
+## Installing WSL
```
Here it would be good to link to - https://splashkit.io/installation/windows-wsl/ which has the full WSL and splashkit guide.

breezy-codes commented on src/content/docs/Products/SplashKit/Splashkit Expansion/Documentation/onboarding-guide.mdx:
```diff
@@ -0,0 +1,205 @@
+on how to install WSL: [Install WSL](https://learn.microsoft.com/en-us/windows/wsl/install). This
+guide has been tested with the default Ubuntu distribution, but others may also work.
+
+### Installing Visual Studio Code
```
same in this section - https://splashkit.io/installation/windows-wsl/step-3/

breezy-codes commented on src/content/docs/Products/SplashKit/Splashkit Expansion/Documentation/onboarding-guide.mdx:
```diff
@@ -0,0 +1,205 @@
+
+Now that WSL is fully configured, it is time to install the splashkit-core repository.
+
+### Forking splashkit-core Repository
```
Here you could possibly link to this as a general guide for forking - https://thoth-tech.netlify.app/products/splashkit/splashkit-website/onboarding/01-setting-up/

breezy-codes commented on src/content/docs/Products/SplashKit/Splashkit Expansion/Documentation/onboarding-guide.mdx:
```diff
@@ -0,0 +1,205 @@
+multiple smaller commits with meaningful descriptions rather than a single monolithic commit. In
+addition, pushing commits to GitHub provides a layer of backup in case of local machine failure.
+
+### Creating a Pull Request
```
Here could include this link - https://thoth-tech.netlify.app/products/splashkit/splashkit-website/onboarding/03-pull-request/

breezy-codes commented on src/content/docs/Products/SplashKit/Splashkit Expansion/Documentation/onboarding-guide.mdx:
```diff
@@ -0,0 +1,205 @@
+fork, and your personal fork. During trimester, PRs should be made against the thoth-tech fork. The
+PR template provides a framework for how to structure the associated PR documentation.
+
+### Responding to Peer Reviews
```
Here could like to this - https://thoth-tech.netlify.app/products/splashkit/splashkit-website/onboarding/05-peer-review/

breezy-codes commented on src/content/docs/Products/SplashKit/Splashkit Expansion/Documentation/onboarding-guide.mdx:
```diff
@@ -0,0 +1,205 @@
+If changes are requested during a PR review, pushing further commits to the same branch will
+automatically be added to the PR.
+
+## Troubleshooting
```
This could link to - https://thoth-tech.netlify.app/products/splashkit/splashkit-website/onboarding/02-troubleshooting/

breezy-codes commented on src/content/docs/Products/SplashKit/Splashkit Expansion/Documentation/onboarding-guide.mdx:
```diff
@@ -0,0 +1,205 @@
+## Troubleshooting
+
+Solutions for common issues can be found below.
+
```
maybe add in one more section on planner board etiquette which can link to - 
https://thoth-tech.netlify.app/products/splashkit/splashkit-website/onboarding/07-planner-board/

breezy-codes commented on src/content/docs/Products/SplashKit/Splashkit Expansion/Documentation/onboarding-guide.mdx:
```diff
@@ -0,0 +1,205 @@
+### Installing Git
+
+Now Git must be installed to WSL. Follow the Microsoft installation instructions:
+[Install Git](https://learn.microsoft.com/en-us/windows/wsl/tutorials/wsl-git).
```
Here you could also add a link to - https://education.github.com/git-cheat-sheet-education.pdf

breezy-codes reviewed (changes requested):
Overall this is a great addition to have as an onboarding guide for splashkit expansion. However, it could be improved by linking to more documents/pages. 

There is a lot of great onboarding pages under splashkit website in thoth tech, that really should be at some point moved to being company wide. But linking them into here would be a great addition to have.

Also linking to some of the official install guides on splashkit would help as they cover extra parts. Also adding in the git cheat sheet would help for people not fully familiar.

matt439 commented:
Thanks for the suggestions. I have added links to the official Thoth Tech documentation. I added 2 new sections covering performing peer review and the planner board.

breezy-codes commented on src/content/docs/Products/SplashKit/Splashkit Expansion/Documentation/onboarding-guide.mdx:
```diff
@@ -0,0 +1,229 @@
+page, keep the default name and click ‘Create fork’.
+
+The following guide for the SplashKit Website has useful information regarding forks and branches
+which can be adjusted for splashkit-core by substitutung repository names:
```
Spelling error "substituting"

breezy-codes commented on src/content/docs/Products/SplashKit/Splashkit Expansion/Documentation/onboarding-guide.mdx:
```diff
@@ -0,0 +1,229 @@
+
+Now that a new branch is created and active, development can begin.
+
+### Building the Test Programs
```
I think it might also be good here to mention the contributing file, and that it has information for these commands in other operating systems (mac and linux)

breezy-codes reviewed (changes requested):
Just a couple more things, but other than that. This is a very good guide for people to get started on.

The only other potential thing, where you've done WSL, you could say that splashkit has install guides for mac and linux which will help you get set up if on other operating systems. It hasn't been merged yet but the new install guide on splashkit for linux has multiple distro options. Its just that this guide is very focused on windows, which generally isnt a problem as most people are on windows. But those couple notes help in the scenarios where someone isnt.

matt439 commented:
I added a link to the CONTRIBUTING.md file for macOS and Linux users. Also fixed the type.

breezy-codes reviewed (approved):
This looks great now, it will be a really helpful piece of documentation to have

ShaunR1991 reviewed (approved):
This is looking great after the changes/suggestions from Brianna. A much needed addition to the onboarding hub. It is very detailed, has great instructions getting set up and love the idea of having troubleshooting for common issues. The instructions are clear and easy to follow, logically placed and is great that it links to some of the other resources in the hub (promotes that cross-team interaction that we want from the start). 

A few very minor typos:
- ettique → etiquette
- contructive feedback → constructive feedback
